### ISSUE TITLE
[Developer Activities]: 2023 The inaugural Developer Activities

### ISSUE THREAD

**Issue by pengtianyue025** [cubefs/cubefs#1920]
## Welcome to  The inaugural Developer Activities

We have selected some topics. You can choose the relevant theme based on your interests and then comment on the selected issue ID.

### Tool Optimization

|Topic|Issue|Module|Assigne|PR|Progress|
|---|---|---|---|---|---|
|[**Easy**] The cfs-cli tool supports QoS management|#1893|cfs-cli||
|[**Easy**] The cfs-cli supports setting the maximum number of dp on datanode|#1897|cfs-cli||
|[**Easy**] The blobstore-cli supports setting background task switch|#1898|blobstore-cli||
|[**Easy**] The blobstore-cli supports offline specified nodes|#1899|blobstore-cli||
|[**Easy**] The blobstore-cli supports showing disk drop or disk repair progress|#1900|blobstore-cli||
|[**Easy**] The blobstore-cli supports Clustermgr to switch masters|#1901|blobstore-cli||
|[**Easy**] The cfs-cli supports viewing all master address information|#1902|cfs-cli||
|[**Medium**] The cfs-cli displays disk information of DataNode|#1895|cfs-cli||
|[**Medium**] The cfs-cli supports querying file information through inodes|#1896|cfs-cli||

### Code Optimization

|Topic|Issue|Module|Assigne|PR|Progress|
|---|---|---|---|---|---|
|[**Easy**] Unit Test Supplement|#1903|Master,MetaNode,DataNode,ObjectNode||
|[**Easy**] Log flush output before crash|#1910|Master,MetaNode,DataNode,ObjectNode||
|[**Easy**] Correctly displays socket and pipe type files|#1501|MetaNode||
|[**Easy**] Support the master to uniformly configure the push gateway address of the data reported by the client|#1913|Blobstore||
|[**Easy**] The RPC module of the erasure code system supports downloading crc data integrity verification|#1914|Blobstore||
|[**Easy**] Master/DataNode/MetaNode start log optimization|#1917|Master,MetaNode,DataNode||
|[**Medium**] Client mount point supports non-repeatable mount|#1904|Cilent||
|[**Medium**] MetaNode requests Master dp information list reuse|#1908|MetaNode||
|[**Medium**] Process (all) log storage rolls according to directory space|#1909|All||
|[**Medium**] The erasure code subsystem supports reporting indicators in Consul mode|#1911|Blobstore||
|[**Medium**] Support the master to uniformly configure the push gateway address of the data reported by the client|#1912|Master||
|[**Medium**] Limit the number of threads for blobnode and datanode to read and write disk|#1915|BlobNode,DataNode||
|[**Medium**] Master does snapshot optimization|#1916|Master||
|[**Medium**] Master qosClientId allocation optimization|#1918|Master||
|[**Medium**] Volume supports sleep capability|#1919|Master||
|[**Hard**] DataNode receiving processing supports speed limit|#1906|DataNode||
|[**Hard**] Clear the abnormal Data Partition of the replica volume|#1907|DataNode,MetaNode,Master||

### Deployment Optimization

|Topic|Issue|Module|Assigne|PR|Progress|
|---|---|---|---|---|---|
|[**Medium**] Automatic production of rpm packages supports version release|#1905|All||

**Comment by pengtianyue025** [comment]
Hi, @rogerogers

Welcome aboard! We're looking forward to your contributions.

**Comment by pengtianyue025** [comment]
> Hello, i want to try #1893

Hi, @rogerogers

Welcome aboard! We're looking forward to your contributions.

**Comment by NaturalSelect** [comment]
Hello, I want to try #1897

**Comment by jjyaoao** [comment]
Hello, I want to try #1903

**Comment by pengtianyue025** [comment]
> [NaturalSelect](/NaturalSelect)

Hi，@NaturalSelect you're welcome!

Understood, please feel free to ask any questions you may have.

**Comment by NaturalSelect** [comment]
> > Hello, I want to try #1897
> 
> Hi，@NaturalSelect you're welcome!
> 
> Understood, please feel free to ask any questions you may have.

Thank you

**Comment by pengtianyue025** [comment]
> Hello, I want to try #1897

Hi, @jjyaoao 

This is an interesting task, maybe you can first familiarize yourself with your favorite modules and then start enjoying coding.

**Comment by pengtianyue025** [comment]
Due to the wide range of topics covered by #1903,  we do not limit the number of applicants who can sign up.

**Everyone is welcome to apply.**

**Comment by xiaodi22013** [comment]
hello, I want to try [#1902 ](url)

**Comment by mfordjody** [comment]
hello, I want to try https://github.com/cubefs/cubefs/issues/1904

**Comment by aFlyBird0** [comment]
Hello, I'm interested in #1899 , thank you!

**Comment by cloorc** [comment]
Hello, I'm going to try #1898 , thanks!

**Comment by pengtianyue025** [comment]
> hello, I want to try [#1902 ](url)
Hi, @xiaodi22013
We look forward to your joining.

**Comment by pengtianyue025** [comment]
> hello, I want to try #1904
Hi, @mfordjody
Welcome to join us, it looks like a good choice.

**Comment by pengtianyue025** [comment]
> Hello, I'm interested in #1899 , thank you!

Hi, @aFlyBird0

Start enjoying your CubeFS journey now.

**Comment by pengtianyue025** [comment]
> Hello, I'm going to try #1898 , thanks!

Hi，@cloorc

You are now in charge of this topic.

**Comment by setcy** [comment]
Hello, I want to try #1900 , thanks!

**Comment by pengtianyue025** [comment]
> Hello, I want to try #1900 , thanks!

Hi, @setcy

Let's embark on a coding journey!

**Comment by 201341** [comment]
hi, #1501 is ok?

**Comment by z1cheng** [comment]
Hello @pengtianyue025, could I try #1914?

**Comment by sjcsjc123** [comment]
Hello, @pengtianyue025 , I want to try #1901 , thanks

**Comment by pengtianyue025** [comment]
> hi, #1501 is ok?

Hi, @201341 
Just do it!

**Comment by pengtianyue025** [comment]
> Hello, @pengtianyue025 , I want to try #1901 , thanks

Hi, @sjcsjc123 

Now you are the owner of #1901

**Comment by pengtianyue025** [comment]
> /assign #1501 Let me do it, thank you. @pengtianyue025

Hi, @rayowang 

I am so sorry, this issue has assigned to @201341 , would you mind picking a different one

**Comment by setcy** [comment]
Hello, can I apply for #1919 as the second topic?

**Comment by pengtianyue025** [comment]
> Hello, can I apply for #1919 as the second topic?

Yes, no problem！ If you have any question, please contact @Victor1319

**Comment by cessory** [comment]
> Hello, I want to try #1915 .

ok, no problem. please contact @cessory

**Comment by NaturalSelect** [comment]
> > Hello, I want to try #1915 .
> 
> ok, no problem. please contact @cessory

Thank you

**Comment by bboyCH4** [comment]
> hello, i want to try #1895 , but maybe i need more context about this topic

Welcome to join us. The cfs-cli tool needs to display the list of disks  one the datanode as well as the list of datapartitions of the disk , more dispaly information details you can contact with me

**Comment by gueFDF** [comment]
Hello, I want to try https://github.com/cubefs/cubefs/issues/1910

**Comment by true1064** [comment]
> 
Hi，@gueFDF you're welcome!

https://github.com/cubefs/cubefs/issues/1910 is Assigned to you .
please feel free to ask me if you have any questions.

**Comment by true1064** [comment]
> Hello, I want to try #1910

Hi, [gueFDF](https://github.com/gueFDF),
https://github.com/cubefs/cubefs/issues/1910 is assigned to you.
If you have any question, please contact me @true1064

**Comment by sjcsjc123** [comment]
Hello, the issue #1901 has been finished, i now choose #1896 , have already create pull request

**Comment by NaturalSelect** [comment]
Hello, I'm interested in #1922 .

**Comment by bboyCH4** [comment]
> Hello, I'm interested in #1922 .
ok, #1922  is assigned to you.If you have any questions,  you can contact with me

**Comment by NaturalSelect** [comment]
> > Hello, I'm interested in #1922 .
> > ok, #1922  is assigned to you.If you have any questions,  you can contact with me

Thank you.

**Comment by true1064** [comment]
> Hello, the issue #1901 has been finished, i now choose #1896 , have already create pull request

#1896 is assigned to you. 
You can keep asking me if you have any questions。

**Comment by NaturalSelect** [comment]
Hello, I just opened a [pull request](https://github.com/cubefs/cubefs/pull/2097) for #1922, and I'm edge to try topic #1918.

**Comment by fengshunli** [comment]
Hello, I want to try https://github.com/cubefs/cubefs/issues/1914

**Comment by Zhytou** [comment]
Hi, I wanna try https://github.com/cubefs/cubefs/issues/1913

**Comment by sejust** [comment]
> Hello, I want to try #1914

Sorry, The issue has some incompleteness and is closed for now. You can try others.

**Comment by sejust** [comment]
> Hi, I wanna try #1913

@Zhytou  Sorry, It seems that someone else has finished this, see #2126.  Many other issues are waiting for you.

**Comment by Monkeyman520** [comment]
Hello！Can I try the issue #1908 ?

**Comment by true1064** [comment]
> Hello！Can I try the issue #1908 ?

@baijiaruo plz check this out
